server/src/http_server/pages/blog/md/html.rs

```diff
@@ -296,45 +274,45 @@ impl IntoHtml for Emphasis {
     }
 }
 
+struct RenderingImage<'img> {
+    image: &'img Image,
+    context: MarkdownRenderContext,
+}
+
+impl<'img> LinkTo for RenderingImage<'img> {
+    fn relative_link(&self) -> String {
+        let RenderingImage { context, image } = self;
+
+        PathBuf::new()
+            .join(&context.current_article_path)
+            .join(&image.url)
+            .to_string_lossy()
+            .to_string()
+    }
+}
+
 impl IntoHtml for Image {
     fn into_html(self, config: &AppConfig, context: &MarkdownRenderContext) -> Result<Markup> {
-        let article_path = context
-            .current_article_path
-            .as_deref()
-            .unwrap_or("unknown");
-
-        let base_url = PathBuf::from(config.base_url.trim_end_matches('/'));
-        let image_path = PathBuf::from(self.url.trim_start_matches("./"));
```

Comment:
`./` means that we stay in the current path so this is actually fine to keep in!

The `PathBuf::join` method know hows to handle this and strips it our for us!

Reply:
You can also call `join` and pass it a &str so you don't have to convert it to a PathBuf first!